assets/src/components/lateView.tsx

```diff
@@ -276,13 +273,19 @@ const LateView = (): ReactElement<HTMLElement> => {
   const lateGhosts = unsortedLateGhosts.sort(compareGhosts)
 
   const lateBuses = vehiclesOrGhosts
-    .filter(isVehicle)
-    .filter((vehicle) => vehicle.scheduleAdherenceSecs >= lateBusThreshold)
-    .sort((a, b) => b.scheduleAdherenceSecs - a.scheduleAdherenceSecs)
+    .filter(isVehicleInScheduledService)
+    .filter(
+      (vehicle) =>
+        vehicle.scheduleAdherenceSecs !== null &&
+        vehicle.scheduleAdherenceSecs >= lateBusThreshold
+    )
+    .sort(
+      (a, b) => (b.scheduleAdherenceSecs || 0) - (a.scheduleAdherenceSecs || 0)
+    )
 
-  const lateVehiclesAndGhosts = (lateGhosts as VehicleOrGhost[]).concat(
-    lateBuses
-  )
+  const lateVehiclesAndGhosts = (
+    lateGhosts as (VehicleInScheduledService | Ghost)[]
```

Comment:
Ah, thanks! I _think_ using the spread operator instead of concat would resolve that typing issue, though I'm having trouble finding documentation about that